src/javafundamentals/streams/Ex02StreamSimpleOperations.java

```diff
@@ -1,10 +1,23 @@
 package javafundamentals.streams;
 
+import java.util.function.Supplier;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
 /**
  * Create an stream of ints with values from 1 to 100
  * (tip: there is a nice method inside IntStream)
  * Print the min, max and count from the stream.
  */
 
 public class Ex02StreamSimpleOperations {
+    public static void main(String[] args) {
+        //IntStream integerStream = IntStream.rangeClosed(1,100);
+
+        Supplier<IntStream> intStreamSupplier = () -> IntStream.rangeClosed(1,100);
+
+        System.out.println(intStreamSupplier.get().min().getAsInt());
```

Comment:
Lets imagine that the numbers are `random` instead of a `rangeClosed`.
with your code, can the `min()` be higher than the `max()`?

Reply:
Its a tricky question, find me in the class to answer to this before merging!